File: src/test/java/dev/kensa/parse/CachingFieldAccessorTest.java
package dev.kensa.parse;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Set;

import static org.assertj.core.api.Assertions.assertThat;

class CachingFieldAccessorTest {

    private CachingFieldAccessor accessor;

    @BeforeEach
    void setUp() {
        accessor = new CachingFieldAccessor(new TestInstance(), Set.of("foo", "boo", "noField"));
    }

    @Test
    void canAccessFieldValues() {
        assertThat(accessor.valueOf("foo")).contains("fooValue");
        assertThat(accessor.valueOf("boo")).contains("booValue");
        assertThat(accessor.valueOf("zoo")).isEmpty();
        assertThat(accessor.valueOf("noField")).isEmpty();
    }

    private static class TestInstance {
        private String foo = "fooValue";
        private String boo = "booValue";
    }
}

File: src/test/java/dev/kensa/parse/CachingFieldAccessorTest.kt
package dev.kensa.parse

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CachingFieldAccessorTest {
    private lateinit var accessor: CachingFieldAccessor

    @BeforeEach
    internal fun setUp() {
        accessor = CachingFieldAccessor(TestInstance(), setOf("foo", "boo", "noField"))
    }

    @Test
    internal fun canAccessFieldValues() {
        assertThat(accessor.valueOf("foo")).isEqualTo("fooValue")
        assertThat(accessor.valueOf("boo")).isEqualTo("booValue")
        assertThat(accessor.valueOf("zoo")).isNull()
    }

    private class TestInstance {
        private val foo = "fooValue"
        private val boo = "booValue"
    }
}